R/Rules-methods.R

```diff
@@ -773,6 +773,59 @@ setMethod(
     }
 )
 
+
+# maxDose-IncrementsHSRBeta ----
+
+#' @rdname maxDose
+#'
+#' @description Determine the maximum possible dose for escalation.
+#'
+#' @aliases maxDose-IncrementsHSRBeta
+#' @example examples/Rules-method-maxDose-IncrementsHSRBeta.R
+#' @export
+setMethod(
+  "maxDose",
+  signature = signature(
+    increments = "IncrementsHSRBeta",
+    data = "Data"
+  ),
+  definition = function(increments, data, ...) {
+    # summary of observed data per dose level
+    y <- factor(data@y, levels = c("0", "1"))
+    dlt_tab <- table(y, data@x)
+
+    # ignore placebo if applied
+    if (data@placebo == TRUE & min(data@x) == data@doseGrid[1]) {
+      dlt_tab <- dlt_tab[, -1]
+    }
+
+    # extract dose names as these get lost if only one dose available
+    doses <- as.numeric(colnames(dlt_tab))
```

Comment:
how about something more explicit, like `unique(sort(data@x))`?

Reply:
`data@x` may contain placebo. Therefore the `dlt_tab` is used. In the step before, placebo is removed if applied. I thought the columns are always sorted, but probably it is better sort and use unique. Will add that and chage doses to non_plcb_doses.